lib/db/DB.js

```diff
@@ -37,10 +34,16 @@ class DB {
 
   async init() {
     try {
+      const dboptions = { ...this.options };
+      dboptions.database = '';
+      const sequelize = new Sequelize(dboptions);
+      sequelize.query(`CREATE DATABASE IF NOT EXISTS ${this.options.database};`);
```

Comment:
This needs `await` before the query and authenticate lines since these are asynchronous methods that must get a response from the db. Also, you can do this without needing to create a new `dboptions` object, see my example below. I tested this locally and it worked.

```javascript
const initSequelize = new Sequelize('', this.sequelize.options.username, this.sequelize.options.password, this.sequelize.options);
await initSequelize.authenticate();
await initSequelize.query(`CREATE DATABASE IF NOT EXISTS ${this.sequelize.options.database};`);
```

Reply:
Since `this.sequelize` is initiated and used in `this.importModels()` it is throwing an error even before reaching the init method. Cloud please guide me on this?

Reply:
The last two lines of the constructor should be:

```javascript
this.sequelize = new Sequelize(options);
this.importModels();
```

Then in `init()` you should start with the three lines from my snippet above. Let me know if that's still not working for you.